xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxUtils.cpp

```diff
@@ -36,13 +36,17 @@ void CDVDDemuxUtils::FreeDemuxPacket(DemuxPacket* pPacket)
 {
   if (pPacket)
   {
-    try {
-      if (pPacket->pData) _aligned_free(pPacket->pData);
-      delete pPacket;
-    }
-    catch(...) {
-      CLog::Log(LOGERROR, "%s - Exception thrown while freeing packet", __FUNCTION__);
+    if (pPacket->pData)
+      _aligned_free(pPacket->pData);
+    if (pPacket->iSideDataElems)
+    {
+      AVPacket avPkt;
+      av_init_packet(&avPkt);
+      avPkt.side_data = reinterpret_cast<AVPacketSideData*>(pPacket->pSideData);
+      avPkt.side_data_elems = pPacket->iSideDataElems;
+      av_packet_free_side_data(&avPkt);
```

Comment:
Don't we have to set pPacket->iSideDatalElems to 0 too?

Reply:
and src ptr might be null - which directly segfaults: https://ffmpeg.org/doxygen/3.1/avpacket_8c_source.html#l00531

Reply:
I think the reinterpret_cast is wrong here and we end up with a dangeling ptr.